### PR TITLE
Add samples in Canvas.drawImage

### PR DESCRIPTION
**Pull request by mateusfccp**
This PR adds two samples to `Canvas.drawImage`.

Solves https://github.com/flutter/flutter/issues/78087.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

### CODE REVIEW

**Review by chinmaygarde — CHANGES_REQUESTED** [review]
This should probably go in flutter/flutter as it references classes and routines not present in this engine vended package. Plus, if this in the framework, the commented code will be tested and is easier to put inside generated documentation.

**Comment by mateusfccp** [comment]
@chinmaygarde The original PR was in `flutter/flutter` repository, but @Hixie suggested me to provide this documentation at the level of `Canvas.drawImage` itself, which is in this repository... (see https://github.com/flutter/flutter/issues/78087#issuecomment-810441639)

Now I am a little confused, did I miss something?

**Comment by Hixie** [comment]
cc @gspencergoog the docs features like sample code and stuff work in dart:ui docs too, no?

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4287]: you'll want to use the `{@tool` stuff to mark up sample code. cc @gspencergoog is that documented anywhere?

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4370]: Avoid "Note..." (see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose)

File: lib/ui/painting.dart
Comment (gspencergoog) [reply lib/ui/painting.dart:4287]: Yes. https://github.com/flutter/flutter/blob/master/dev/snippets/README.md

**Comment by gspencergoog** [comment]
@Hixie Yes, it gets generated as part of generating docs for the framework that is using that engine revision.

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4483]:
```suggestion
  /// See also:
  ///
  ///  * [ImageProvider], which identifies an image and provides its stream.
  ///  * [createLocalImageConfiguration], which provides an [ImageConfiguration]
  ///    accordingly to the current context.
```

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4308]: Might be good to put a bit more detail here explaining what the limitations of this approach are, explaining that it's specific to when you're using widgets, and so on

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4314]: constructor before properties please

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4321]: comments should end with a period

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4337]: "Each time a new frame is provided" sounds vaguely like it refers to frames of the application. Maybe "Each time a new image frame is provided (e.g. for an animated GIF)" would be clearer?

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4354]: avoid else after return

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4379]: constructor first

File: lib/ui/painting.dart
Comment (Hixie) [inline lib/ui/painting.dart:4397]: consider putting the first of these two paragraphs as part of the description before the first example (at the top of the @tool) and the second paragraph at the top of the second @tool (maybe along with more details there too)

**Comment by Hixie** [comment]
@chinmaygarde I'll leave this up to you for a more detailed review. If there's a better place to put this then that's fine. The problem we're trying to resolve here is that people look at the API docs for `canvas.drawImage`, and they ask the question "how do I render an asset using this", and right now we provide no guidance whatsoever. I'm happy for the answer to be something that doesn't involve widgets if you prefer that. We should definitely have more sample code in the API docs for Canvas though. :-)

**Review by flar — CHANGES_REQUESTED** [review]
Per my comment in the issue - wouldn't a better place to describe how to interface the low level rendering APIs defined in `dart:ui` with the higher level Flutter asset management system be a class that specifically bridges the gap - such as CustomPaint/CustomPainter?

**Comment by mateusfccp** [comment]
@flar

@chinmaygarde said the same. Maybe you are right? @Hixie suggested me first to open a PR here, so I did, but I don't mind changing it to `flutter/flutter`.

As a side-note, `Canvas.drawAtlas` also does have reference to the `CustomPainter` in the provided sample, so maybe you would like to change it too: https://api.flutter.dev/flutter/dart-ui/Canvas/drawAtlas.html

**Comment by zanderso** [comment]
Until https://github.com/flutter/flutter/issues/80144 is addressed, we should hold off on adding more sample code in the engine repo.

**Comment by chinmaygarde** [comment]
Closing this for now.

**Pull request closed without merging**
